### ISSUE TITLE
False positive in unneeded_override

### ISSUE THREAD

**Issue by dalemyers** [realm/SwiftLint#5355]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

In the following code, the override is deemed redundant, even though it provides an default parameter:

```swift
class Foo {
    func bar(value: String) {
        print(value)
    }
}

class Baz: Foo {
    override func bar(value: String = "Hello") {
        super.bar(value: value)
    }
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
[snipped]/test.swift:8:5: warning: Unneeded Overridden Functions Violation: Remove overridden functions that don't do anything except call their super (unneeded_override)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.54.0
* Installation method used: CocoaPods
* Paste your configuration file: N/A
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations): No
* Which Xcode version are you using (check `xcodebuild -version`): Xcode 15.0.1 Build version 15A507
* Do you have a sample that shows the issue? See above
